Remove go-lxc dependencies

masnax opened the pull request:
Reimplements SetupTrust from LXD so we don't import too much from LXD. I think this is a simple enough function that we can duplicate it, not sure if we should bloat LXD's shared package with it.

tomponline commented on microcloud/service/lxd_join.go:
```diff
@@ -0,0 +1,75 @@
+	}
+
+	err = target.CreateCertificate(post)
+	if err != nil && err.Error() != ErrCertificateExists.Error() {
```
Does the lxd server response when certificate already exists use a specific error code as ideally we would detect it using api.StatusErrorCheck() here.

masnax replied:
Actually, looking into this more it looks like this error is long gone.

masnax replied:
Actually not, apparently I was grepping incorrectly :)

masnax replied:
OK, I've updated the error to use `409 Conflict` if a certificate already exists.

tomponline reviewed (changes requested):
Pr description is a little confusing as go-lxc is the go wrapper of liblxc not lxd's go client :)
Or is this an indirect dependency you're trying to avoid?

masnax commented:
> Pr description is a little confusing as go-lxc is the go wrapper of liblxc not lxd's go client :) Or is this an indirect dependency you're trying to avoid? Does this also need a go mod update?

Yeah I'm trying to avoid bringing in liblxc.

tomponline commented:
Thanks, ive merged the lxd pr now.

stgraber commented:
@masnax I take it this needs an update for the LXD change as well as including a gomod update?

masnax commented:
@stgraber I've updated this now for the LXD changes. Also since the LXD changes include the new init types, I've gone ahead and added a commit changing the initialization to use the `lxd init --preseed` command, as discussed with @tomponline

tomponline reviewed (approved):
Lgtm

tomponline commented:
Good to see all of those indirect dependencies removed.

masnax commented:
Just fixed a small typo (The preseed yaml was both the last argument and being written to stdin)
